game-core/src/main/java/games/strategy/triplea/delegate/battle/steps/change/ClearGeneralCasualties.java

```diff
@@ -0,0 +1,35 @@
+package games.strategy.triplea.delegate.battle.steps.change;
+
+import static games.strategy.triplea.delegate.battle.BattleStepStrings.REMOVE_CASUALTIES;
+
+import games.strategy.engine.delegate.IDelegateBridge;
+import games.strategy.triplea.delegate.ExecutionStack;
+import games.strategy.triplea.delegate.battle.BattleActions;
+import games.strategy.triplea.delegate.battle.BattleState;
+import games.strategy.triplea.delegate.battle.steps.BattleStep;
+import java.util.List;
+import lombok.AllArgsConstructor;
+
+@AllArgsConstructor
+public class ClearGeneralCasualties implements BattleStep {
+  private static final long serialVersionUID = 995118832242624142L;
+
+  protected final BattleState battleState;
```

Comment:
Why protected on these variables? Can we mark them as private?

> Effective Java Item 17: Design and document for inheritance or else prohibit it
https://medium.com/@rufuszh90/effective-java-item-17-design-and-document-for-inheritance-or-else-prohibit-it-be6041719fbc